server/src/main/java/org/elasticsearch/common/ReferenceDocs.java

```diff
@@ -46,6 +46,28 @@ public enum ReferenceDocs {
     CONCURRENT_REPOSITORY_WRITERS,
     ARCHIVE_INDICES,
     HTTP_TRACER,
+    HEAP_SIZE_CHECK,
```

Comment:
Naming nit: rather than a `_CHECK` suffix could we use a prefix and call these `BOOTSTRAP_CHECK_HEAP_SIZE` and so on? I think that'd make it a bit clearer to future readers.